applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -75,60 +79,87 @@ public static Builder builder() {
         return service.setKey(serviceEndpoint, result.key(), result.label(), result, null, getETagValue(ETAG_ANY));
     }
 
     /**
-     * Creates or updates a configuration value in the service.
+     * Creates or updates a configuration value in the service with the given key. Partial updates are not supported.
+     *
+     * @param key The key for the configuration setting to create or update.
+     * @param value The value of this configuration setting.
+     * @return ConfigurationSetting that was created or updated.
+     * @throws IllegalArgumentException If {@code key} or {@code value} are {@code null} or an empty string.
+     */
+    public Mono<Response<ConfigurationSetting>> setSetting(String key, String value) {
+        return setSetting(new ConfigurationSetting().key(key).value(value));
+    }
+
+    /**
+     * Creates or updates a configuration value in the service. Partial updates are not supported.
      *
      * <p>
-     * If {@link ConfigurationSetting#etag()} is specified, the configuration value is updated if the current setting's
+     * If {@link ConfigurationSetting#etag() etag} is specified, the configuration value is updated if the current setting's
      * etag matches. If the etag's value is equal to {@link ConfigurationClient#ETAG_ANY}, the setting will always be
      * updated.
      *
-     * <p>
-     * The label value for the ConfigurationSetting is optional.
-     *
      * @param setting The configuration setting to create or update.
      * @return ConfigurationSetting that was created or updated.
      * @throws NullPointerException If {@code setting} is {@code null}.
-     * @throws IllegalArgumentException If {@link ConfigurationSetting#key()} is {@code null} or an empty string.
-     * @throws ServiceRequestException If the {@link ConfigurationSetting#etag()} was specified, is not
-     * {@link ConfigurationClient#ETAG_ANY}, and the current configuration
-     * value's etag does not match.
+     * @throws IllegalArgumentException If {@link ConfigurationSetting#key() key} or {@link ConfigurationSetting#value() value}
+     * are {@code null} or an empty string.
+     * @throws ServiceRequestException If the {@link ConfigurationSetting#etag() etag} was specified, is not
+     * {@link ConfigurationClient#ETAG_ANY}, and the current configuration value's etag does not match.
      */
     public Mono<Response<ConfigurationSetting>> setSetting(ConfigurationSetting setting) {
-        ConfigurationSetting result = validateSetting(setting);
+        ConfigurationSetting result = validateSetting(setting, true);
 
         return service.setKey(serviceEndpoint, result.key(), result.label(), result, getETagValue(result.etag()), null);
     }
 
+    /**
+     * Updates an existing configuration value in the service with the given key. The setting must already exist.
+     *
+     * <p>
+     * Partial updates are not supported.
+     *
+     * @param key The key for the configuration setting to update.
```

Comment:
```suggestion
     * @param key The key of the configuration setting to update.
```

Reply:
I'm not sure if it should be 'for' or 'of', or if I should just not care. Consider it here and elsewhere, and do whatever you feel is better English.

Reply:
That's a great question. Ahaha. I asked the internet. https://www.ecenglish.com/learnenglish/lessons/prepositions-for-of-and-to

Reply:
The important question is: did you pass the test? :-)